File: backend/src/main/java/com/xyehyin/hexuanning/entity/Course.java
package com.xyehyin.hexuanning.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 课程实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "course")
@EntityListeners(AuditingEntityListener.class)
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(length = 50)
    private String subject;

    @Column(length = 20)
    private String grade;

    @Column(length = 20)
    private String semester;

    @Column(nullable = false)
    @Builder.Default
    private Boolean enabled = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private User teacher;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(length = 255)
    private String coverImage;

    @Column
    private Integer totalHours;

    @Column
    @Builder.Default
    private Integer completedHours = 0;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    @Column(length = 20)
    private CourseStatus status = CourseStatus.DRAFT;

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createTime;

    @LastModifiedDate
    private LocalDateTime updateTime;

    // 添加与知识点的一对多关系，配置级联删除
    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<KnowledgePoint> knowledgePoints;

    /**
     * 课程状态枚举
     */
    public enum CourseStatus {
        DRAFT,      // 草稿
        PUBLISHED,  // 已发布
        COMPLETED,  // 已完成
        ARCHIVED    // 已归档
    }
}

File: frontend/src/api/types/course.ts
import type { ApiResponse, PageResult } from './common'
import type { CategoryFlatVO } from './category'
import type { UserSimpleVO, UserVO } from './user'

export type CourseStatus = 'DRAFT' | 'PUBLISHED' | 'COMPLETED' | 'ARCHIVED'

export interface CourseVO {
  id: number
  name: string
  description: string
  subject: string
  grade: string
  semester?: string
  enabled: boolean
  coverImage?: string
  totalHours?: number
  completedHours?: number
  status: CourseStatus
  teacherName?: string
  teacherId?: number
  categoryName?: string
  categoryId?: number
  createTime: string
  updateTime: string
}

export interface CourseDetailVO {
  id: number
  name: string
  description: string
  subject: string
  grade: string
  semester?: string
  enabled: boolean
  coverImage?: string
  totalHours?: number
  completedHours?: number
  status: CourseStatus
  teacher?: UserVO
  category?: CategoryFlatVO
  createTime: string
  updateTime: string
  progress: number
}

export interface CourseSimpleVO {
  id: number
  name: string
  subject: string
  grade: string
  semester: string
  status: CourseStatus
  teacher: UserSimpleVO
}

export interface CreateCourseDTO {
  name: string
  description: string
  subject: string
  grade: string
  semester?: string
  enabled?: boolean
  categoryId?: number
  coverImage?: string
  totalHours?: number
  status?: CourseStatus
}

export interface UpdateCourseDTO {
  name?: string
  description?: string
  subject?: string
  grade?: string
  semester?: string
  enabled?: boolean
  categoryId?: number
  coverImage?: string
  totalHours?: number
  completedHours?: number
  status?: CourseStatus
}

export interface CourseStatisticsVO {
  totalCourses: number
  publishedCourses: number
  draftCourses: number
  archivedCourses: number
  myCourses: number
  myPublishedCourses: number
}

export interface SubjectVO {
  id: number
  name: string
  code: string
}

export interface GradeVO {
  id: number
  name: string
  code: string
}

export type PageResultCourseVO = PageResult<CourseVO>

export type ApiResponseCourseVO = ApiResponse<CourseVO>
export type ApiResponseCourseDetailVO = ApiResponse<CourseDetailVO>
export type ApiResponseListCourseVO = ApiResponse<CourseVO[]>
export type ApiResponsePageResultCourseVO = ApiResponse<PageResultCourseVO>
export type ApiResponseCourseStatisticsVO = ApiResponse<CourseStatisticsVO>
export type ApiResponseListSubjectVO = ApiResponse<SubjectVO[]>
export type ApiResponseListGradeVO = ApiResponse<GradeVO[]>
